src/Domain/Model/Resource.php

```diff
@@ -11,15 +11,19 @@ class Resource
     private $type;
     private $description;
     private $authorId;
+    private $votesTotal;
+    private $votes;
 
-    public function __construct($type, $url, $description, $authorId)
+    public function __construct($type, $url, $description, $authorId, $votesTotal = 0, $votes = array())
```

Comment:
constructors should consist only from mandatory parameters for the object to exist. Everything else should be changed via settes.

you can also set the `votesTotal` and `votes` to it's defaults in the body